assets/src/edit-story/app/media/utils/useFFmpeg.js

```diff
@@ -45,38 +45,123 @@ const isDevelopment = process.env.NODE_ENV === 'development';
  */
 const isFileTooLarge = ({ size }) => size >= MEDIA_TRANSCODING_MAX_FILE_SIZE;
 
-function useTranscodeVideo() {
+/**
+ * Custom hook to interact with FFmpeg.
+ *
+ * @see https://ffmpeg.org/ffmpeg.html
+ *
+ * @return {{
+ * isFeatureEnabled: boolean,
+ * isFileTooLarge: (function(File): boolean),
+ * isTranscodingEnabled: boolean,
+ * canTranscodeFile: (function(File): boolean),
+ * transcodeVideo: (function(File): Promise<File>)
+ * getFirstFrameOfVideo: (function(File): Promise<File>)
+ * }} Functions and vars related to FFmpeg usage.
+ */
+function useFFmpeg() {
   const { ffmpegCoreUrl } = useConfig();
   const {
     state: { currentUser },
   } = useCurrentUser();
+  /**
+   * Whether the video optimization flag is enabled.
+   *
+   * @type {boolean} Whether the feature flag is enabled.
+   */
   const isFeatureEnabled = useFeature('videoOptimization');
 
+  async function getFFmpegInstance(file) {
+    const { createFFmpeg, fetchFile } = await import(
+      /* webpackChunkName: "chunk-ffmpeg" */ '@ffmpeg/ffmpeg'
+    );
+
+    const ffmpeg = createFFmpeg({
+      corePath: ffmpegCoreUrl,
+      log: isDevelopment,
+    });
+    await ffmpeg.load();
+
+    ffmpeg.FS('writeFile', file.name, await fetchFile(file));
+
+    return ffmpeg;
+  }
+
   /**
-   * Transcode a video using ffmpeg.
+   * Extract a video's first frame using FFmpeg.
    *
-   * @param {File} file File object of original video file.
-   * @return {Promise<File>} File object of transcoded video file.
+   * Exact seeking is not possible in most formats, so ffmpeg will seek to the closest seek point before position.
+   *
+   * @param {File} file Original video file object.
+   * @return {Promise<File>} File object for the video frame.
    */
-  async function transcodeVideo(file) {
+  async function getFirstFrameOfVideo(file) {
     //eslint-disable-next-line @wordpress/no-unused-vars-before-return
-    const trackTiming = getTimeTracker('load_video_transcoding');
+    const trackTiming = getTimeTracker('load_video_poster_ffmpeg');
 
     try {
-      const { createFFmpeg, fetchFile } = await import(
-        /* webpackChunkName: "chunk-ffmpeg" */ '@ffmpeg/ffmpeg'
+      const ffmpeg = await getFFmpegInstance(file);
+
+      const tempFileName = uuidv4() + '.jpeg';
+      const outputFileName = getFileName(file) + '.jpeg';
+
+      await ffmpeg.run(
+        // Desired position.
+        // Using as an input option (before -i) saves us some time by seeking to position.
+        '-ss',
+        '00:00:01.000',
```

Comment:
Should this time match, [this](https://github.com/google/web-stories-wp/blob/ffe309081f0f58c44dac47b856c9e15149e3ba5a/assets/src/edit-story/app/media/utils/getFirstFrameOfVideo.js#L40-L46).

Reply:
Both FFmpeg and browsers will only take it as a hint and might adjust the number. Anything around the 1s mark is OK, so no need for changing.